### PR TITLE
Implement fetch_bundle for TDR plugin (#1844)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**

### CODE REVIEW

**Comment by noah-aviel-dove** [comment]
Rebasing on #1857 to address review feedback that affects shared commits. This may somewhat invalidate peer review.

**Comment by noah-aviel-dove** [comment]
Reminder to self: fix manifest `size` for data files before requesting Hannes review.

**Comment by noah-aviel-dove** [comment]
Re-requesting peer review due to substantial changes following rebase and addition of file descriptors.

File: src/azul/bigquery.py [hunk 1]
```diff
@@ -28,6 +29,10 @@ def assert_table_exists(self, dataset_name: str, table_name: str) -> None:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def create_table(self, dataset_name: str, table_name: str, schema: JSONs, rows: JSONs) -> None:
```
Comment (achave11) [inline src/azul/bigquery.py:33]: Is this missing  docstrings as per the other two methods in this class?

**Comment by noah-aviel-dove** [comment]
> Regarding the changes in general, it seems that there is string usage in both single quotes and double quotes, did you apply any specific JSON/str/dict standards or was it an overlook?

I _tried_ to apply a specific standard but overthought it and kept changing my delimiters until I gave up on consistency. Simplified and fixed now.

**Pull request merged**